lib/logstorage/storage.go

```diff
@@ -365,6 +365,60 @@ func (s *Storage) PartitionSnapshotList() []string {
 	return snapshotPaths
 }
 
+// PartitionSnapshotDelete removes the snapshot with the given name if it belongs to an active partition.
+//
+// The snapshotName must match the directory name returned by PartitionSnapshotCreate() or PartitionSnapshotList().
+func (s *Storage) PartitionSnapshotDelete(snapshotName string) error {
+	if snapshotName == "" {
+		return fmt.Errorf("snapshot name cannot be empty")
+	}
+	if err := snapshotutil.Validate(snapshotName); err != nil {
+		return fmt.Errorf("invalid snapshot name %q: %w", snapshotName, err)
+	}
+
+	s.partitionsLock.Lock()
+	ptws := append([]*partitionWrapper{}, s.partitions...)
+	for _, ptw := range ptws {
+		ptw.incRef()
+	}
+	s.partitionsLock.Unlock()
+
+	var targetPtw *partitionWrapper
+
+	for _, ptw := range ptws {
+		ptPath := ptw.pt.path
+		snapshotsDir := filepath.Join(ptPath, snapshotsDirname)
+		snapshotPath := filepath.Join(snapshotsDir, snapshotName)
+		if !fs.IsPathExist(snapshotPath) {
```

Comment:
This logic looks too fragile. It may result in the deletion of the snapshot from unexpected per-day partition if multiple partitions have snapshots with the same name.

It is better from maintainability and usability to pass the `path=/full/path/to/the/snapshot` query arg to `/internal/partition/snapshot/delete`. Then users can copy&paste the full path returned from `/internal/partition/snapshot/create` or `/internal/partition/snapshot/list` to the `path` query arg without deciding on which part of the path must be copied to the `name` query arg.

The code must verify that the passed absolute path belongs to one of the active partition before deleting the snapshot at the given path.